src/main/java/ua/softserveinc/tc/dao/impl/DayOffDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package ua.softserveinc.tc.dao.impl;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import ua.softserveinc.tc.constants.DayOffConstants;
+import ua.softserveinc.tc.dao.DayOffDao;
+import ua.softserveinc.tc.entity.DayOff;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaDelete;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.time.LocalDate;
+import java.util.List;
+
+import static ua.softserveinc.tc.util.DateUtil.toDate;
+
+@Repository
+public class DayOffDaoImpl extends BaseDaoImpl<DayOff> implements DayOffDao {
```

Comment:
We must begin transaction before send request to database, and we must close such transaction. We van make it by hands by TransactionManager, but in Spring we have annotation @Transactional.